src/Session/Session.cc

```diff
@@ -722,15 +722,21 @@ void Session::OnAddRequiredTiles(const CARTA::AddRequiredTiles& message, int z,
                 const auto& encoded_coordinate = message.tiles(i);
                 auto tile = Tile::Decode(encoded_coordinate);
                 auto raster_tile_data = Message::RasterTileData(file_id, sync_id, animation_id);
-
+                bool tile_error(false);
                 if (_frames.count(file_id) && _frames.at(file_id)->FillRasterTileData(raster_tile_data, tile, requested_z, stokes,
-                                                  compression_type, compression_quality, is_current_z)) {
+                                                  compression_type, compression_quality, is_current_z, tile_error)) {
                     // Only use deflate on outgoing message if the raster image compression type is NONE
                     SendFileEvent(
                         file_id, CARTA::EventType::RASTER_TILE_DATA, 0, raster_tile_data, compression_type == CARTA::CompressionType::NONE);
                 } else {
-                    spdlog::warn(
-                        "Discarding stale tile request for channel={}, x={}, y={}, layer={}", requested_z, tile.x, tile.y, tile.layer);
+                    if (tile_error) {
+                        SendLogEvent(fmt::format("Invalid mip calculation: channel={}, x={}, y={}, layer={}", requested_z, tile.x, tile.y,
+                                         tile.layer),
+                            {"invalid mip calculation"}, CARTA::ErrorSeverity::WARNING);
```

Comment:
Minor nitpick: this should be something more general, like `animation`. We haven't been consistent about this elsewhere.